applications/simple_pdw_pipeline/siggen.c

```diff
@@ -0,0 +1,364 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <unistd.h>
+#include <stdio.h>
+#include <sys/socket.h>
+#include <pthread.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <getopt.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+#define SIGNAL_PACKET_SIZE 8192
+#define HEADER_SIZE 1
+#define COMPLEX_SIZE 2
+#define BUFFER_SIZE (HEADER_SIZE + (SIGNAL_PACKET_SIZE * COMPLEX_SIZE))
+#define NUMBER_OF_BUFFERS 2
+
+#define NUMBER_OF_ITERATIONS 15
+
+int16_t main_buffer[BUFFER_SIZE*NUMBER_OF_BUFFERS];
+bool valid_buffer[NUMBER_OF_BUFFERS];
+pthread_mutex_t mutexen[NUMBER_OF_BUFFERS];
+pthread_cond_t signal[NUMBER_OF_BUFFERS];
+bool quit;
+
+struct siggen_config{
+  //Number of bits that are error
+  unsigned int error_bits;
+  // 0 to RAND_MAX
+  unsigned int probability_of_transmission;
+  unsigned int offset;
+  unsigned int increment;
+};
+
+struct transmit_config{
+  // UDP port
+  unsigned short port;
+  // IP address to transmit to
+  char * address;
+};
+
+struct chirp_description{
+  //0 to 2^15 - 1
+  int16_t initial_delta;
+  //0 to 2^31 - 1
+  int32_t chirp_rate;
+  // bits to attenuate signal by
+  unsigned char attenuation;
+  //Bin start
+  unsigned long start;
+  //Bin length
+  unsigned long length;
+};
+
+struct live_signal{
+  bool valid;
+  struct chirp_description chirp;
+};
+
+/*
+The fpsin and fpcos functions were from 
+https://www.nullhardware.com/blog/fixed-point-sine-and-cosine-for-embedded-systems/
+and have the following copywright:
+==========LICENCE=============
+Copyright 2018 Andrew Steadman
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://nam11.safelinks.protection.outlook.com/?url=http%3A%2F%2Fwww.apache.org%2Flicenses%2FLICENSE-2.0&data=05%7C01%7Cjjomier%40nvidia.com%7C5992447d975048be2a7508db67525601%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C638217375574205616%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=bHeZqmTlmFx5BCcketj0j%2FTod6eoeTOrkouAU%2FAojFY%3D&reserved=0
```

Comment:
I would suggest to remove this section since the license is the same as the current file (Apache 2) but put the following at line 4:

The fpsin and fpcos functions are Copyright 2018 Andrew Steadman and available at:
https://www.nullhardware.com/blog/fixed-point-sine-and-cosine-for-embedded-systems/

Reply:
Excellent. Sorry, didn't know what the format and didn't want to butcher other's words when licences were involved! Will revise.